packages/shared/src/components/squads/PublicSquadSubmissionActions.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactElement, useCallback } from 'react';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
+import { squadsPublicGuide } from '../../lib/constants';
+import { anchorDefaultRel } from '../../lib/strings';
+import { SquadStatus } from './settings';
+import { TimerIcon } from '../icons';
+import { usePublicSquadRequests } from '../../hooks';
+import { useLazyModal } from '../../hooks/useLazyModal';
+import { PUBLIC_SQUAD_REQUEST_REQUIREMENT } from '../../lib/config';
+import { LazyModal } from '../modals/common/types';
+import { Squad } from '../../graphql/sources';
+
+interface PublicSquadSubmissionActionsProps {
+  squad: Squad;
+  isDetailsVisible?: boolean;
+}
+
+const PublicSquadSubmissionActions = (
+  props: PublicSquadSubmissionActionsProps,
+): ReactElement => {
+  const { squad, isDetailsVisible = true } = props;
+  const { isFetched, status } = usePublicSquadRequests({
+    sourceId: squad?.id,
+  });
+  const { openModal } = useLazyModal();
+  const postsCount = squad?.flags?.totalPosts ?? 0;
+  const isEligible = postsCount >= PUBLIC_SQUAD_REQUEST_REQUIREMENT;
+
+  const onSubmit = useCallback(() => {
+    if (!squad.id) {
```

Comment:
this means squad can be null?
Maybe we should make squad optional in the props then already?

Reply:
I'm removing since it should not be optional, copy-paste went wrong for this ahahha
Thanks for the feedback! :(